### ISSUE TITLE
[FEATURE] choose one group when using group share level

### ISSUE THREAD

**Issue by lianneli** [apache/incubator-kyuubi#3482]
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### Describe the feature

When a user has several groups in ugi, user can choose a group needed when using group share level. 

### Motivation

Group level is more efficient than user level, with less time of resource apply and less parameters to optimize. For user, engines for different groups mean different queues and different resources. Now the group is chosen automatically by header group in ugi, which may not flexible for user, as well as the platform in upper level cannot provide a better suggestion for user.

### Describe the solution

em...may be I can solve this problem by using parameters transferred by jdbc connection  

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

**Comment by bowenliang123** [comment]
I am afraid that choosing group name by connection param, without appropriate authentication or management, may cause privilige leaking, group level resource control leaking and etc.

**Comment by pan3793** [comment]
Does Hadoop have the primary group concept?

**Comment by bowenliang123** [comment]
Another user-group binding in my mind is #3308 , fetching UserStore from Ranger which contains user to group(s) relations. But authz is spark engine side plugin.

**Comment by pan3793** [comment]
We can extract the `GroupProvider` traits and provide different mechanism

**Comment by bowenliang123** [comment]
Yes , continue to allow choosing group name via params and cheking with different mechanisms with LDAP、Kerberos、JDBC 、Ranger UserStore and etc to restrict allowed groups.

**Comment by lianneli** [comment]
I thought so once, but get groups from Kerberos or Ranger UserStore makes it more complexible for Kyuubi server. Maybe enhance the `org.apache.kyuubi.service.authentication.PasswdAuthenticationProvider` is more effective. For my scene, select one group name from UGI GroupNames through JDBC is enough.

**Comment by HaoYang670** [comment]
I guess this [PR](https://github.com/apache/kyuubi/pull/4822) can solve the issue. cc @lianneli @pan3793
